src/v3/aes.rs

```diff
@@ -0,0 +1,89 @@
+use super::{verify_signature, IV_LEN};
+use crate::{
+    aes::{decrypt_attached_document_core, EncryptionKey, PlaintextDocument},
+    EncryptedPayload, Error, MAGIC,
+};
+use protobuf::Message;
+
+// [3, b"IRON", <2 bytes of header len>]
+const DETACHED_HEADER_LEN: usize = 7;
+
+const V3: u8 = 3u8;
+
+/// Decrypt a V3 detached document and verifies its signature.
+pub fn decrypt_detached_document(
+    key: &EncryptionKey,
+    payload: EncryptedPayload,
```

Comment:
This type isn't actually correct. In my upcoming PR I'm moving this type into the V5 module since it says on it that it starts with `0IRON`. We should make our own type for V3.

I'm trying to make these types more correct by construction instead of them just being new types that don't actually verify anything about their contents. If we did that here, you could verify that the `len > DETATCHED_HEADER_LEN + IV_LEN` on construction and then in decrypt you wouldn't need to check it again. I've also been experimenting with splitting the bytes up front on construction, but we can discuss that if you're interested.

One last thing: I put the decrypt function directly on the EncryptedPayload type, though I'm still struggling with naming. 

Let's talk this morning and we can align our work.

Reply:
I move all the validation into the TryFrom. I didn't move the decrypt on the EncryptedPayload yet, so let me know if that's still what you prefer